From 9ef40d75e78c28dbe2c7e64e4aaf7298efb07735 Mon Sep 17 00:00:00 2001
From: Philip Rebohle <philip.rebohle@tu-dortmund.de>
Date: Wed, 21 Sep 2022 14:39:29 +0200
Subject: [PATCH] vkd3d: add storage-buffer descriptors to local pools

VKD3D uses storage-buffer descriptors for internal operations, including
query-meta resolve layouts. Reserve them in every local descriptor pool.
---
 libs/vkd3d/command.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libs/vkd3d/command.c b/libs/vkd3d/command.c
index 68fa3bc8..38f1b0c1 100644
--- a/libs/vkd3d/command.c
+++ b/libs/vkd3d/command.c
@@ -1324,6 +1324,7 @@ static VkDescriptorPool d3d12_command_allocator_allocate_descriptor_pool(
         {VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1024},
         {VK_DESCRIPTOR_TYPE_UNIFORM_TEXEL_BUFFER, 1024},
         {VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE, 1024},
+        {VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, 1024},
         {VK_DESCRIPTOR_TYPE_STORAGE_TEXEL_BUFFER, 1024},
         {VK_DESCRIPTOR_TYPE_STORAGE_IMAGE, 1024},
         /* must be last in the array */
-- 
2.39.5

